src/test/java/com/exasol/adapter/dialects/saphana/util/IntegrationTestSetup.java

```diff
@@ -0,0 +1,189 @@
+        }
+    }
+
+    private static void uploadVsJarToBucket(final Bucket bucket) {
+        try {
+            bucket.uploadFile(PATH_TO_VIRTUAL_SCHEMAS_JAR, VIRTUAL_SCHEMAS_JAR_NAME_AND_VERSION);
+        } catch (final FileNotFoundException | BucketAccessException | TimeoutException exception) {
+            throw new IllegalStateException("Failed to upload jar " + PATH_TO_VIRTUAL_SCHEMAS_JAR + " to "
+                    + VIRTUAL_SCHEMAS_JAR_NAME_AND_VERSION, exception);
+        }
+    }
+
+    private String getTestHostIpFromInsideExasol() {
+        final Map<String, ContainerNetwork> networks = this.exasolContainer.getContainerInfo().getNetworkSettings()
+                .getNetworks();
+        if (networks.size() == 0) {
+            return null;
+        }
+        return networks.values().iterator().next().getGateway();
+    }
+
+    private AdapterScript createAdapterScript(final ExasolSchema schema) {
+        final String content = "%scriptclass com.exasol.adapter.RequestDispatcher;\n" //
+                + "%jar /buckets/bfsdefault/default/" + VIRTUAL_SCHEMAS_JAR_NAME_AND_VERSION + ";\n" //
+                + "%jar /buckets/bfsdefault/default/drivers/jdbc/" + JDBC_DRIVER_NAME + ";\n";
+        return schema.createAdapterScript(ADAPTER_SCRIPT_EXASOL, AdapterScript.Language.JAVA, content);
+    }
+
+    public VirtualSchema createVirtualSchema(final Schema hanaSchema) {
+        final Map<String, String> properties = new HashMap<>(Map.of("CATALOG_NAME", hanaSchema.getName()));
+        properties.putAll(debugProperties());
+        return this.exasolFactory.createVirtualSchemaBuilder("HANA_VIRTUAL_SCHEMA_" + (this.virtualSchemaCounter++))
+                .adapterScript(this.adapterScript).connectionDefinition(this.connectionDefinition)
+                .sourceSchemaName(hanaSchema.getName()).properties(properties).build();
+    }
+
+    private Map<String, String> debugProperties() {
+        final String debugAddress = System.getProperty("com.exasol.virtualschema.debug.address");
+        if (debugAddress == null) {
+            return Collections.emptyMap();
+        }
+        final String logLevel = System.getProperty("com.exasol.virtualschema.debug.level");
+        return Map.of("DEBUG_ADDRESS", debugAddress, "LOG_LEVEL", (logLevel != null ? logLevel : "ALL"));
+    }
+
+    public HanaSchema createHanaSchema() {
+        final String newSchemaName = "HANA_SCHEMA_" + hanaSchemaCounter++;
+        if (hanaSchemaExists(newSchemaName)) {
+            executeInHana("DROP SCHEMA " + newSchemaName + " CASCADE");
+        }
+        return hanaFactory.createSchema(newSchemaName);
+    }
+
+    public void clean(final Schema hanaSchema) {
+        if (hanaSchemaExists(hanaSchema.getName())) {
+            hanaSchema.drop();
+        }
+    }
+
+    private boolean hanaSchemaExists(final String schemaName) {
+        try (ResultSet rs = executeInHana("SELECT COUNT(*) FROM SYS.SCHEMAS WHERE SCHEMA_NAME='" + schemaName + "'")) {
```

Comment:
```suggestion
        try (ResultSet rs = executeInHana("SELECT COUNT(1) FROM SYS.SCHEMAS WHERE SCHEMA_NAME='" + schemaName + "'")) {
```

Reply:
`SELECT 1 ... WHERE ... LIMIT 1` would be more efficient.

Reply:
✅